### ISSUE TITLE
Add "Usage" section to README

### ISSUE THREAD

**Issue by cap10morgan** [tianon/pgp-happy-eyeballs#1]
I'm having a hard time making this work, and I think I would benefit greatly from having a "Usage" section in the README. Ideally with both Docker-based and non-Docker-based examples. Thanks!

**Comment by tianon** [comment]
Took a first pass in 1a3cddd, hopefully that's helpful?

**Comment by cap10morgan** [comment]
Yep, that's a good start, thanks! I think the missing pieces now are:

1. What is the actual Docker command you use to run this once rawdns is configured and running?
1. Docker config details. I assume you need to point the daemon's DNS at 127.0.0.1 for this to work with things like `docker build`?

**Comment by tianon** [comment]
> 1. What is the actual Docker command you use to run this once rawdns is configured and running?

https://github.com/tianon/pgp-happy-eyeballs/blob/1a3cddd23404582bfc85515d518e55a6cf193225/hack-my-builds.sh#L69

(Essentially just `docker run` with an appropriate `--name` to reference from my `rawdns.json` and `--dns` to make sure it talks to the right DNS servers and doesn't get hijacked to try routing to itself.)

> 2. Docker config details. I assume you need to point the daemon's DNS at 127.0.0.1 for this to work with things like `docker build`?

That's a little bit obscured in https://github.com/tianon/pgp-happy-eyeballs/blob/1a3cddd23404582bfc85515d518e55a6cf193225/hack-my-builds.sh#L24-L40, but here's what my current local machine's got which will probably help:

```json
{
	"storage-driver": "overlay2",
	"bip": "172.18.0.1/16",
	"dns": [
		"172.18.0.1",
		"1.1.1.1",
		"1.0.0.1"
	]
}
```

(Using `bip` to ensure my `docker0` bridge has a consistent static IP I can bind my `rawdns` container to in order to point `dns` at the bridge IP so that other containers can actually hit it with fallbacks for containers that aren't on the default bridge network.)
